recipe/meta.yaml

```diff
@@ -89,6 +89,11 @@ outputs:
     files:
       - bin/curl               # [unix]
       - Library/bin/curl.exe*   # [win]
+    build:
+      ignore_run_exports: 
+        - libssh2          # [linux or osx  or win]
+        - krb5             # [linux or osx or win]
+        - zlib             # [linux or win]
```

Comment:
This seems highly suspicious. All three are listed as `host` requirements, and `libssh2` is an explicitly stated `run` requirement. The fact that we have then to ignore the run exports suggests that either (1) they are not actually needed because all of the functionality is in libcurl, or (2) the configure/build steps are doing something wrong (e.g., statically linking or not detecting the presence of these libraries during configuration).

If these run exports do need to be ignored, the recipe should contain an explanation as to why (beyond "to get the build to work").

Reply:
I think changes in the last versions can partially explain it (see changes in 7.80.0 https://curl.se/changes.html):
- `krb5-config` was replaced by `pkg-config` https://github.com/curl/curl/pull/7916 
- `Get the libssh2 version at runtime if possible ` https://github.com/curl/curl/pull/7768 and  `Makefile.m32: fix to not require OpenSSL with options -libssh2 and -rtmp` https://github.com/curl/curl/pull/7895
- I didn't find mentions about `zlib` in the last versions

Reply:
Added a comment about why we use ignore_run_exports in 7.80.0

Reply:
Looking at those notes and the generated `bin/curl` executable, the correct solution is not to use `ignore_run_exports` but to remove zlib, libssh2, and krb5 from the `host` and `run` requirements (along with `ignore_run_exports`).

The functionality `bin/curl` needs is actually provided by the libcurl DSO, and everything else is redundant.

```
$ readelf -d  tmp/contents/bin/curl

Dynamic section at offset 0x319a0 contains 27 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libcurl.so.4]
 0x0000000000000001 (NEEDED)             Shared library: [libpthread.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x000000000000000f (RPATH)              Library rpath: [$ORIGIN/../lib]
```

Reply:
There is an error on `osx64 `and `osx-arm64` if we remove from `host `and `run `zlib, libssh2, and krb5:
```
   INFO: sysroot: '/opt/MacOSX10.10.sdk/' files: '['usr/share/man/mann/zero.n', 'usr/share/man/mann/yencode.n', 'usr/share/man/mann/yaml.n', 'usr/share/man/mann/xsxp.n']'
   INFO (curl,bin/curl): Needed DSO lib/libcurl.4.dylib found in output-artifacts::libcurl-7.80.0-h6dfd666_0
  ERROR (curl,bin/curl): Needed DSO lib/libz.1.dylib found in ['zlib']
  ERROR (curl,bin/curl): .. but ['zlib'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (curl,bin/curl): Needed DSO /opt/MacOSX10.10.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
overlinking check failed 
["  ERROR (curl,bin/curl): .. but ['zlib'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)"]
```

Reply:
To fix this error I added to `run`:
`        - zlib  # [osx]`

Reply:
@chenghlee Is it correct to use `zlib # [osx or s390x]` in run to fix the above error

Reply:
A big maybe for osx, but the fact that linux-s390x links to zlib but the other linux-* platforms do not seems like a red flag to me. I can't think of a good reason why curl on s390x would require zlib, but curl x86_64 and ARM on do not; that suggests that something is going wrong in either the configure and/or link steps. I need to wrap up a few things, but I can take a look later this week.

Reply:
Marking as resolved; see comment from @adipietro18 below.